Src/FluentAssertions/Specialized/AssemblyAssertions.cs

```diff
@@ -139,6 +141,64 @@ public AndConstraint<AssemblyAssertions> Reference(Assembly assembly, string bec
         return new AndWhichConstraint<AssemblyAssertions, Type>(this, foundType);
     }
 
+    /// <summary>Asserts that the assembly is unsigned.</summary>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because"/>.
+    /// </param>
+    public AndConstraint<AssemblyAssertions> BeUnsigned(string because = "", params object[] becauseArgs)
+    {
+        Execute.Assertion
+            .BecauseOf(because, becauseArgs)
+            .ForCondition(Subject.GetName().GetPublicKey() is not { Length: > 0 })
+            .FailWith(
+                "Expected assembly {0} to be unsigned{reason}, but it is.", Subject.FullName);
+
+        return new(this);
+    }
+
+    /// <summary>Asserts that the assembly has the specified public key.</summary>
+    /// <param name="publicKey">
+    /// The base-16 string representation of the public key.
+    /// </param>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because"/>.
+    /// </param>
+    /// <exception cref="ArgumentNullException"><paramref name="publicKey"/> is <see langword="null"/>.</exception>
+    /// <exception cref="ArgumentException"><paramref name="publicKey"/> is empty.</exception>
+    public AndConstraint<AssemblyAssertions> HavePublicKey(string publicKey, string because = "", params object[] becauseArgs)
+    {
+        Guard.ThrowIfArgumentIsNullOrEmpty(publicKey);
+
+        var bytes = Subject.GetName().GetPublicKey();
+        var assemblyKey = BitConverter.ToString(bytes).Replace("-", string.Empty, StringComparison.Ordinal);
+
+        Execute.Assertion
+            .BecauseOf(because, becauseArgs)
+            .ForCondition(bytes.Length != 0)
+            .FailWith(
+                "Expected assembly {0} to have public key {1}{because}, " +
+                "but it unsigned.",
+                Subject.FullName, publicKey);
+
+        Execute.Assertion
+            .BecauseOf(because, becauseArgs)
+            .ForCondition(assemblyKey == publicKey)
+            .FailWith(
+                "Expected assembly {0} to have public key {1}{because}, " +
+                "but it has {2} instead.",
+                Subject.FullName, publicKey, assemblyKey);
```

Comment:
```suggestion
        Execute.Assertion
            .BecauseOf(because, becauseArgs)
            .WithExpectation("Expected assembly {0} to have public key {1}{because}, ", Subject.FullName , publicKey)
            .ForCondition(bytes is not null)
            .FailWith("but public key's bytes are null.")
            .Then
            .ForCondition(bytes.Length != 0)
            .FailWith("but it unsigned.")
            .Then
            .ForCondition(assemblyKey == publicKey)
            .FailWith("but it has {0} instead.", assemblyKey)
            .Then
            .ClearExpectation();
```

Reply:
You can concat the assertions you avoid to repeat yourself.